src/modules/Edition/index.ts

```diff
@@ -3,18 +3,31 @@ import { ethers, BigNumber, Contract } from "ethers";
 import DCNT721A from '../../contracts/DCNT721A.json';
 import { MetadataRendererInit } from '../MetadataRenderer';
 
+export type TokenGateConfig = {
+  tokenAddress: string;
+  minBalance: number;
+  saleType: number;
+}
+
 const deploy = async (
   sdk: SDK,
   name: string,
   symbol: string,
+  adjustableCap: boolean,
   maxTokens: number,
   tokenPrice: BigNumber,
   maxTokenPurchase: number,
+  presaleStart: number,
+  presaleEnd: number,
+  saleStart: number,
```

Comment:
need `saleEnd` too right? 

Or, is there a reason we are offering less feature-full experience than Sound & Zora who currently offer `publicSaleEnd`?